vault/auth.go

```diff
@@ -224,6 +224,13 @@ func (c *Core) loadCredentials() error {
 
 		// Upgrade to table-scoped entries
 		for _, entry := range c.auth.Entries {
+			// The auth backend "aws-ec2" was named "aws" in the master.
+			// This is to support upgrade procedure from "aws" to "aws-ec2".
+			if entry.Path == "aws/" {
+				entry.Path = "aws-ec2/"
```

Comment:
Unfortunately we can't change the path because if the path is wrong the expiration manager won't be able to find the revocation function.

So I think this needs to check entry.Type rather than entry.Path in the `if` clause, and *only* update `entry.Type`. That should make sure that the correct backend is loaded, but keep the path the same.